File: android/app/src/main/java/com/dconeybe/countapi/CountApiDemo.java
package com.dconeybe.countapi;

import static com.dconeybe.countapi.AggregateField.count;
import static com.dconeybe.countapi.AggregateField.min;
import static com.dconeybe.countapi.AggregateField.sum;

public final class CountApiDemo {

  private CountApiDemo() {
  }

  public static void countDemo(Query query) {
    AggregateQuerySnapshot snapshot = query.count().get(AggregateSource.SERVER).getResult();
    System.out.println("Found " + snapshot.getCount() + " documents");
  }

  public static void aggregateDemo(Query query) {
    AggregateQuery aggregateQuery = query.aggregate(count(), min("age"), sum("salary"));
    AggregateQuerySnapshot snapshot = aggregateQuery.get().getResult();

    Object minAge = snapshot.get(min("age"));
    long minAgeStr = (minAge instanceof Long) ? (Long) minAge : 0;
    Double totalSalary = snapshot.getDouble(sum("salary"));
    double totalSalaryDbl = (totalSalary instanceof Double) ? (Double) totalSalary : Double.NaN;

    System.out.println(
        "Found " + snapshot.getCount() + " people,"
        + " the youngest being " + minAgeStr
        + " earning a total of $" + totalSalaryDbl
    );
  }

}

File: android/app/src/main/java/com/dconeybe/countapi/CountApiDemo.kt
package com.dconeybe.countapi

import com.dconeybe.countapi.AggregateField.*

internal fun countDemoKt(query: Query) {
  val snapshot = query.count().get(AggregateSource.SERVER).result
  println("Found " + snapshot.count + " documents")
}

internal fun aggregateDemoKt(query: Query) {
  val aggregateQuery = query.aggregate(count(), min("age"), sum("salary"))
  val snapshot = aggregateQuery.get().result

  val minAge: Long = snapshot.getField<Long>(min("age")) ?: 0
  val totalSalary: Double = snapshot.get(sum("salary")) ?: 0.0

  println(
    "Found ${snapshot.count} people, the youngest being $minAge earning a total of $totalSalary"
  )
}